Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,685 @@
+                        testRun = testrunApi.Create((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testRun, testSuite.Id, "test-suite");
+                    }
+                }
+                else
+                {
+                    //##update existing TestSuite -> TestRun
+                    foreach (QtestTest test in mappedTestSuite.Tests)
+                    {
+                        QTestApiModel.TestRunWithCustomFieldResource testRun = testrunApi.Get((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), (long)Convert.ToInt32(test.Runs[0].RunID), "descendents");
+                        QTestApiModel.TestCaseWithCustomFieldResource testCase = testcaseApi.GetTestCase((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testRun.TestCase.Id);
+
+                        ActivitiesGroup ag = businessFlow.ActivitiesGroups.Where(x => (x.ExternalID == test.TestID.ToString())).FirstOrDefault();
+                        if (ag != null)
+                        {
+                            foreach (ActivityIdentifiers actIdent in ag.ActivitiesIdentifiers)
+                            {
+                                if ((actIdent.IdentifiedActivity.ExternalID == null) && (actIdent.IdentifiedActivity.ExternalID2 == null))
+                                {
+                                    QTestApiModel.TestCaseWithCustomFieldResource newTestCase = new QTestApiModel.TestCaseWithCustomFieldResource(null, null, null, new List<QTestApiModel.PropertyResource>());
+                                    newTestCase.Description = actIdent.ActivityDescription;
+                                    newTestCase.Name = actIdent.ActivityName;
+                                    newTestCase = testcaseApi.CreateTestCase((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), newTestCase);
+                                    QTestApiModel.TestStepResource stepResource = new QTestApiModel.TestStepResource(null, null,
+                                                                                                                            ((Activity)actIdent.IdentifiedActivity).Description == null ? string.Empty : ((Activity)actIdent.IdentifiedActivity).Description,
+                                                                                                                            ((Activity)actIdent.IdentifiedActivity).Expected == null ? string.Empty : ((Activity)actIdent.IdentifiedActivity).Expected);
+                                    stepResource.PlainValueText = ((Activity)actIdent.IdentifiedActivity).ActivityName;
+                                    testcaseApi.AddTestStep((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testCase.Id, stepResource);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            // existingActivitiesGroups.Add(ag);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)